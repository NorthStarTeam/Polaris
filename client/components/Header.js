import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import { UserContext } from './AccState';

const Header = () => {
  const userObj = useContext(UserContext);
  const { isLogin } = userObj;

  const headerLeft = () => {
    return (
      <ul className="link-left">
        <li>
          <Link to="/">Polaris</Link>
        </li>
        <li>
          <Link to="/about">About</Link>
        </li>
      </ul>
    );
  };

  if (isLogin) {
    return (
      <div className="div-ul">
        {headerLeft()}
        <ul className="link-right">
          <li>
            <Link to="/logout">LogOut</Link>
          </li>
          <li>
            <Link to="/setting">Settings</Link>
          </li>
        </ul>
      </div>
    );
  } else {
    return (
      <div className="div-ul">
        {headerLeft()}
        <ul className="link-right">
          <li>
            <Link to="/signup">Join</Link>
          </li>
          <li>
            <Link to="/login">Login</Link>
          </li>
        </ul>
      </div>
    );
  }
};

export default Header;

// <div class="card card-body text-center">
//   <h1>
//     <i class="fa fa-sticky-note"></i> MiNote
//   </h1>
//   <p>Create an account or login</p>
//   <a href="/users/register" class="btn btn-primary btn-block mb-2">
//     Register
//   </a>
//   <a href="/users/login" class="btn btn-secondary btn-block">
//     Login
//   </a>

//   <a href="/users/auth/github" class="btn btn-info btn-block">
//     <i class="fa fa-github-alt"></i> Login with Github
//   </a>
//   <a href="/users/auth/fb" class="btn btn-primary btn-block">
//     <i class="fa fa-facebook-official"></i> Login with Facebook
//   </a>
//   <a href="/users/auth/gg" class="btn btn-warning btn-block">
//     <i class="fa fa-facebook-official"></i> Login with Google
//   </a>
// </div>;
