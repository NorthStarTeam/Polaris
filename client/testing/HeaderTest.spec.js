import React from 'react';
import { MemoryRouter } from 'react-router-dom'

import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Header from '../components/Header';

describe('Header', () => {
    
    //snapshot test
    test('Header renders correctly', () => {
        const component = renderer.create(<MemoryRouter> <Header /> </MemoryRouter>)
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot();
    })

    //enzyme tests
    it('renders a div with the class of div-ul', () => {
        const wrapper = mount(<MemoryRouter> <Header /> </MemoryRouter>)
        expect(wrapper.find('div.div-ul').length).toEqual(1)
    })


})

