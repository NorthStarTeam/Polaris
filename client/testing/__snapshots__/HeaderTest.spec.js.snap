// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header renders correctly 1`] = `
Array [
  " ",
  <div
    className="div-ul"
  >
    <ul
      className="link-left"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Polaris
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
    <ul
      className="link-right"
    >
      <li>
        <a
          href="/signup"
          onClick={[Function]}
        >
          Join
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </div>,
  " ",
]
`;
